functorch/test/test_aotdispatch.py

```diff
@@ -731,6 +656,95 @@ def assert_compiler(gm: torch.fx.GraphModule, _):
+    xfail('linalg.cholesky'),
+    skip('msort'),
+    xfail('nn.functional.dropout'),
+    xfail('to_sparse'),
+    xfail('addcdiv'),
+    xfail('cholesky'),
+    xfail('cumulative_trapezoid'),
+    xfail('diag_embed'),
+    xfail('logit'),
+    xfail('trapezoid'),
+    xfail('trapz'),
+    xfail('corrcoef'),
+    xfail('cov'),
+    xfail('chalf'),  # RuntimeError: "sum_cpu" not implemented for 'ComplexHalf'
+    skip('nn.functional.binary_cross_entropy_with_logits'),  # seems to fail sometimes?
+    skip('nn.functional.margin_ranking_loss'),  # seems flaky
+    decorate('matmul', decorator=unittest.skipIf(IS_ARM64, 'flaky')),
+    decorate('__rmatmul__', decorator=unittest.skipIf(IS_ARM64, 'flaky')),
+}
+
+def _test_aot_autograd_helper(self, device, dtype, op):
+    if not op.supports_autograd:
+        self.skipTest("Op does not support autograd")
+
+    sample_inputs_itr = op.sample_inputs(device, dtype, requires_grad=True)
+    for sample_input in sample_inputs_itr:
+        args = [sample_input.input] + list(sample_input.args)
+        kwargs = sample_input.kwargs
+
+        if not all([isinstance(i, torch.Tensor) and i.dtype == torch.float for i in args]):
+            self.skipTest("not all inputs are float tensors")
+
+        def f(args):
+            return op.op(*args, **kwargs)
+
+        t = f(args)
+        if isinstance(t, tuple):
+            self.skipTest("output is a tuple")
+            continue
+
+        def reset_grads():
+            def f(x):
+                x.grad = None
+            pytree.tree_map(f, args)
+
+        def get_grads(args):
+            return pytree.tree_map(lambda x: x.grad, args)
+
+        compiled_f = compiled_function(f, nop, nop)
+
+        reset_grads()
+        compiled_f(args).sum().backward()
+        compiled_grad = get_grads(args)
+
+        reset_grads()
+        f(args).sum().backward()
+        orig_grad = get_grads(args)
+        self.assertEqual(orig_grad, compiled_grad)
+
+        def create_new_arg(x):
+            return x.detach().uniform_(0, 1).requires_grad_(x.requires_grad)
+
+        args = pytree.tree_map(create_new_arg, args)
+
+        reset_grads()
+        compiled_f(args).sum().backward()
+        compiled_grad = get_grads(args)
+
+        reset_grads()
+        f(args).sum().backward()
+        orig_grad = get_grads(args)
+        self.assertEqual(orig_grad, compiled_grad)
+
+class TestEagerFusionOpInfo(AOTTestCase):
+    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @skipOps('TestEagerFusionOpInfo', 'test_aot_autograd_exhaustive', aot_autograd_failures)
+    def test_aot_autograd_exhaustive(self, device, dtype, op):
+        _test_aot_autograd_helper(self, device, dtype, op)
+
+    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @patch("functorch.compile.config.use_dynamic_shapes", True)
+    @patch("functorch.compile.config.use_fake_tensor", True)
+    @patch("functorch.compile.config.use_functionalize", False)
```

Comment:
why is functionalize disabled?

Reply:
Because functionalization currently doesn't work with symbolic shapes, so enabling functionalization just causes all tests to fail. Will add note.